nixos/platform/default.nix

```diff
@@ -188,12 +188,52 @@ in {
       };
     };
 
+    flyingcircus.passwordlessSudoPackages = mkOption {
+      description = ''
+        Works similar to security.sudo.extraRules, but `commands` are
+        interpreted as paths relative to a given `package` so binary can be
+        run from the full Nix store path with `sudo`.
+
+        This option also creates a corresponding rule for
+        `/run/current-system/sw/` to be able to run the globally installed
+        binary from $PATH with `sudo`.
+
+        You still need to add the package to `environment.systemPackages`
+        yourself to make it globally available.
```

Comment:
This approach has 2 pitfalls:

- user error: a user might either forget to add the package there, or add a different package (derivation) of the same tool to `environment.systemPackage` than used in the sudo rule. This can yield unexpected defaults.  
  To protect against this, we could check for the existence of each `package` in the list `environment.systemPackages` and warn/ error if it is not found.
- malicious user: Put a different package into `environment.systemPackages` than intended/ expected by the sudo rule. This allows a malicious user to e.g. run a shell script as a different user then.  
  The previously mentioned existence check is no good countermeasure, because users can add additional packages with a conflicting path but higher `meta.priority` instead to get them included in `/run/current-system/sw`. Additionally, tight now crafting such a config is only possible for users that can modify system config anyways, giving them acces to other code execution means anyways.

----

Given that additional checks are mainly a convenience feature but no good security feature, I am okay with not doing them and keeping it this way.